### PR TITLE
Parallel testing

### PR DESCRIPTION
**Pull request by kacf**

### CODE REVIEW

File: meta-mender-qemu/scripts/mender-qemu [hunk 1]
```diff
@@ -130,8 +136,8 @@ for maybe_kvm in -enable-kvm ""; do
             -m 256M \
             $BOOTLOADER_ARG \
             -net nic,macaddr="$RANDOM_MAC" \
-            -net user,hostfwd=tcp::8822-:22$QEMU_NET_HOSTFWD \
-            -display vnc=:23 \
+            -net user,hostfwd=tcp::${PORT_NUMBER}-:22$QEMU_NET_HOSTFWD \
+            -display vnc=:${VNC_NUMBER} \
```
Comment (oleorhagen) [inline meta-mender-qemu/scripts/mender-qemu:140]: Q: When have we ever used `vnc`? :p

File: tests/acceptance/test_snapshot.py [hunk 2]
```diff
@@ -161,33 +164,34 @@ def test_snapshot_avoid_deadlock(self, connection):
...
-        "terminal", ["", "screen -D -m -L -Logfile screen.log.tmp"]
-    )
+    @pytest.mark.parametrize("terminal", ["", "screen -D -m -L -Logfile %%"])
     def test_snapshot_using_mender_artifact(
         self, terminal, bitbake_path, bitbake_variables, connection
     ):
-        try:
+        with tempfile.NamedTemporaryFile(
+            suffix=".mender"
+        ) as artifact, tempfile.NamedTemporaryFile() as screen_log:
```
Comment (oleorhagen) [inline tests/acceptance/test_snapshot.py:173]: Nice, this was news to me! Cool

File: tests/acceptance/test_snapshot.py [hunk 3]
```diff
@@ -161,33 +164,34 @@ def test_snapshot_avoid_deadlock(self, connection):
...
+            terminal = terminal.replace("%%", screen_log.name)
             (active, _) = determine_active_passive_part(bitbake_variables, connection)
 
             try:
                 subprocess.check_call(
-                    "%s mender-artifact write rootfs-image %s -n test -t test -o test_snapshot_using_mender_artifact.mender -f ssh://%s@%s:%s"
+                    "%s mender-artifact write rootfs-image %s -n test -t test -o %s -f ssh://%s@%s:%s"
```
Comment (oleorhagen) [inline tests/acceptance/test_snapshot.py:179]: Tbh, this would've been a prime candidate for an f-string imo

**Review by oleorhagen — APPROVED** [review]
Seriously fine work! I'm impressed

File: meta-mender-qemu/scripts/mender-qemu (hunk 1 above)
Comment (kacf) [reply meta-mender-qemu/scripts/mender-qemu:140]: I think I have used it exactly once, probably five years ago! 😂

File: tests/acceptance/test_snapshot.py (hunk 3 above)
Comment (kacf) [reply tests/acceptance/test_snapshot.py:179]: I actually started with that, but it became so insanely long that I changed it back! This string is a bit of a catch-22, it's ugly either way, so we can choose between a very long string, or a shorter string with less obvious substitution. In the end I decided to keep it as is, but no strong feelings.

Comment (oleorhagen) [reply tests/acceptance/test_snapshot.py:179]: `catch-22` strings are the worst!

**Comment by kacf** [comment]
Finally https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/683420593 passed. The one failure which is there is some infrastructure error, and happened after all the tests had already passed.

**Pull request merged**